packages/communication-react/review/communication-react.api.md

```diff
@@ -1256,6 +1263,7 @@ export interface SendBoxProps {
     onRenderSystemMessage?: (systemMessage: string | undefined) => React_2.ReactElement;
     onSendMessage?: (content: string) => Promise<void>;
     onTyping?: () => Promise<void>;
+    strings?: Partial<SendBoxStrings>;
```

Comment:
Let's be consistent with how we handle optional strings. Ideally:

* Providing `strings` is always optional.
* When providing `strings`, all fields are optional.

As for how particular strings are made optional, there are two approaches:
* Use `Partial<XStrings>`
* All fields in `XStrings` are always optional.

I personally prefer the latter -- we never want the fields to be required, so the type should itself mark them all as optional.

Reply:
Thanks for pointing this out. The reason why I chose the former which is something like:
``` const interface XProps {
    ...
    strings?: Partial<XStrings>
}
```
so that all string fields are require when defining `ComponentStrings` because it is defined as::
```
export interface ComponentStrings {
    x: XStrings,
    y: YStrings
}
```
The above enforces me to write complete locales.
For example, if I delete string for a component from my locale file like this:
![image](https://user-images.githubusercontent.com/79475487/124329598-8b83e980-db40-11eb-90fe-ec2b3fafae36.png)
I will get the error:
![image](https://user-images.githubusercontent.com/79475487/124329633-9b9bc900-db40-11eb-9ca8-b63e6ffdf6fe.png)
and if I just delete one field from a component like this one:
![image](https://user-images.githubusercontent.com/79475487/124329763-e1f12800-db40-11eb-9b6a-a14a67c2e211.png)
I will get this error:
![image](https://user-images.githubusercontent.com/79475487/124329813-00572380-db41-11eb-9fd1-371775ba8538.png)

But maybe we want developers to provide partial overrides and defaulting the rest to en-US strings. In this case, I agree that all fields in `XStrings` should be optional.

Reply:
iiuc,
* `useLocale` must always provide all strings
* Composite pros or Component props can provide some strings and skip the rest.

This sounds like the right approach to me, and the way you implement that via `Partial` and required fields also looks great. Let's make this consistent (I saw a mix of approaches in the implementation which led me to the initial comment) and document these requiremets in the storybook documentation for localization.